code/datums/components/_component.dm

```diff
@@ -123,10 +123,12 @@
 /datum/component/proc/_RemoveFromParent()
 	var/datum/P = parent
 	var/list/dc = P.datum_components
+	var/list/components_of_type //pre init so we're not creating vars every loop
+	var/list/subtracted
```

Comment:
the list returned by _GetInverseTypeList() might not be large enough on average to justify pre initing these vars

Reply:
eh fuck it i dont want to benchmark it anymore